File: leetcode/js/toeplitz-matrix.js
/**
 * @param {number[][]} matrix
 * @return {boolean}
 */
var isToeplitzMatrix = function (matrix) {
  for (let i = 0; i < matrix[0].length - 1; i++) {
    const elem = matrix[0][i];

    for (let j = 1; j < matrix.length; j++) {
      if (i + j < matrix[0].length && elem !== matrix[j][i + j]) {
        return false;
      }
    }
  }

  for (let i = 1; i < matrix.length - 1; i++) {
    const elem = matrix[i][0];

    for (let j = 1; j < matrix[0].length; j++) {
      if (i + j < matrix.length && elem !== matrix[i + j][j]) {
        return false;
      }
    }
  }

  return true;
};

var isToeplitzMatrix = function (matrix) {
  for (let i = 1; i < matrix.length; i++) {
    for (let j = 1; j < matrix[0].length; j++) {
      if (matrix[i][j] !== matrix[i - 1][j - 1]) {
        return false;
      }
    }
  }

  return true;
};

File: leetcode/rust/toeplitz-matrix.rs
impl Solution1 {
    pub fn is_toeplitz_matrix(matrix: Vec<Vec<i32>>) -> bool {
        for i in 1..matrix.len() {
            for j in 1..matrix[0].len() {
                if (matrix[i][j] != matrix[i - 1][j - 1]) {
                    return false;
                }
            }
        }

        true
    }
}

impl Solution2 {
    pub fn is_toeplitz_matrix(matrix: Vec<Vec<i32>>) -> bool {
        for i in 1..matrix.len() {
            if matrix[i - 1][..matrix[0].len() - 1] != matrix[i][1..] {
                return false;
            }
        }

        true
    }
}
